### PR TITLE
Feature/rules modal

### PR DESCRIPTION
**Pull request by VVMig**

### CODE REVIEW

File: src/app/auth/rules/RulesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Comment (RuslanShohanov) [inline src/app/auth/rules/RulesModal.tsx:1]: Something is wrong with formatting in this file, isn't it?

File: src/packages/components/modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+export const Modal: React.FC<ModalProps> = ({
+    children,
+    className,
+    handleModalClose,
+    stateClass,
+    delay
+}) => {
+    const modalRef = useRef<HTMLDivElement>(null);
+    const [buttonLight, setButtonLight] = useState(false);
```
Comment (RuslanShohanov) [inline src/packages/components/modal/Modal.tsx]: isButtonHighlighted? isButtonActive?

File: src/packages/components/modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        event.target === modalRef.current ||
+        modalRef.current?.contains(event.target as Node);
+
+    const closeModal: React.MouseEventHandler = (event) => {
+        if (!isModalTarget(event)) handleModalClose();
+        event.stopPropagation();
+    };
+
+    const toggleButtonLight: React.MouseEventHandler = (event) => {
+        if (!isModalTarget(event)) setButtonLight(true);
```
Comment (RuslanShohanov) [inline src/packages/components/modal/Modal.tsx]: It's better to use curly braces even if condition body contains only one line. That increases code readability
```
if() {
...
} else {
...
}
```

File: src/packages/components/modal/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            scrollAllow()
+        }
+    })
+
+    const isModalTarget = (event: React.MouseEvent) =>
+        event.target === modalRef.current ||
+        modalRef.current?.contains(event.target as Node);
+
+    const closeModal: React.MouseEventHandler = (event) => {
+        if (!isModalTarget(event)) handleModalClose();
```
Comment (RuslanShohanov) [inline src/packages/components/modal/Modal.tsx]: Here as well

**Review by RuslanShohanov — COMMENTED** [review]
Do you use prettier or something else? Seems you have different formatting for same files (for example `Rules.tsx` and `Modal.tsx`)

File: src/app/auth/rules/RulesModal.tsx (hunk 1 above)
Comment (VVMig) [reply src/app/auth/rules/RulesModal.tsx:1]: I don't see)
What?

Comment (VVMig) [reply src/app/auth/rules/RulesModal.tsx:1]: Semicolon
Prettier auto inserts them

Comment (RuslanShohanov) [reply src/app/auth/rules/RulesModal.tsx:1]: In this file you have tabs that contain 4 spaces I guess
image

And for example in Rules.tsx
image

Try to keep same formatting for all .tsx/.ts files

Comment (VVMig) [reply src/app/auth/rules/RulesModal.tsx:1]: Today work from home and prettier wasn't configure.

**Review by RuslanShohanov — APPROVED** [review]

**Pull request merged**